rust/rsmgp-sys/src/rsmgp.rs

```diff
@@ -0,0 +1,133 @@
+use std::ffi::CStr;
+
+use crate::context::*;
+#[double]
+use crate::mgp::ffi;
+use crate::mgp::*;
+use crate::result::*;
+use crate::value::*;
+use mockall_double::double;
+
+pub struct MgpResultRecord {
+    record: *mut mgp_result_record,
+}
+
+pub fn make_result_record(context: &Memgraph) -> MgpResult<MgpResultRecord> {
+    unsafe {
+        let record = ffi::mgp_result_new_record(context.result());
+        if record.is_null() {
+            return Err(MgpError::UnableToCreateResultRecord);
+        }
+        Ok(MgpResultRecord { record })
+    }
+}
+
+pub fn insert_result_record(
+    mgp_record: &MgpResultRecord,
+    mgp_name: &CStr,
+    mgp_value: &MgpValue,
+) -> MgpResult<()> {
+    unsafe {
+        let inserted =
+            ffi::mgp_result_record_insert(mgp_record.record, mgp_name.as_ptr(), mgp_value.ptr);
+        if inserted == 0 {
+            return Err(MgpError::PreparingResultError);
+        }
+        Ok(())
+    }
+}
+
+pub fn set_memgraph_error_msg(msg: &CStr, context: &Memgraph) {
+    unsafe {
+        let status = ffi::mgp_result_set_error_msg(context.result(), msg.as_ptr());
+        if status == 0 {
+            panic!("Unable to pass error message to the Memgraph engine.");
+        }
+    }
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+pub fn add_read_procedure(
+    proc_ptr: extern "C" fn(*const mgp_list, *const mgp_graph, *mut mgp_result, *mut mgp_memory),
+    name: &CStr,
+    module: *mut mgp_module,
+) -> *mut mgp_proc {
+    unsafe { ffi::mgp_module_add_read_procedure(module, name.as_ptr(), Some(proc_ptr)) }
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+pub fn add_int_result_type(procedure: *mut mgp_proc, name: &CStr) -> MgpResult<()> {
```

Comment:
This, and other result types can be inserted by providing a MgpValue instance to the Record, looks better from API perspective, rather than calling different methods of insertion e.g. `add_int_result_type`, `add_bool_result_type`

Reply:
This is a bit problematic because at the time the result type has to be specified, there is no `MgpValue` (inside `mgp_init_module`). I'll think about how to introduce some "per module" object to hold all registered procedures and types in one place.